src/time_machine/__init__.py

```diff
@@ -351,7 +351,7 @@ def utcnow() -> dt.datetime:
     if not coordinates_stack:
         result: dt.datetime = _time_machine.original_utcnow()
         return result
-    return dt.datetime.utcfromtimestamp(time())
+    return dt.datetime.fromtimestamp(time(), dt.UTC)
```

Comment:
Two fixes required:

1. `dt.UTC` was only added in Python 3.11, for compat we need `dt.timezone.utc`
2. The patched function, `utcnow()`, needs to return a naive datetime, as `utcfromtimestamp()` does. We need an extra `replace()` call to remove `tzinfo`.

```suggestion
    return dt.datetime.fromtimestamp(time(), dt.timezone.utc).replace(tzinfo=None)
```